sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PortEvaluationStage.java

```diff
@@ -158,37 +122,32 @@ protected void setProtos(PodInfoBuilder podInfoBuilder, Protos.Resource resource
         if (getTaskName().isPresent()) {
             String taskName = getTaskName().get();
             Protos.TaskInfo.Builder taskBuilder = podInfoBuilder.getTaskBuilder(taskName);
-            taskBuilder.getCommandBuilder().setEnvironment(
-                    withPortEnvironmentVariable(taskBuilder.getCommandBuilder().getEnvironment(), port));
-
-            if (taskBuilder.hasDiscovery()) {
-                Protos.Ports.Builder portsBuilder = taskBuilder.getDiscoveryBuilder().getPortsBuilder();
-                Optional<Protos.Port.Builder> builderOptional = portsBuilder.getPortsBuilderList().stream()
-                        .filter(b -> b.getName().equals(getPortName()))
-                        .findFirst();
-                if (builderOptional.isPresent()) {
-                    setPortBuilder(builderOptional.get(), port);
-                } else {
-                    portsBuilder.addPorts(makePortBuilder(port));
 
-                }
-            } else {
-                Protos.DiscoveryInfo.Builder discoveryInfoBuilder = Protos.DiscoveryInfo.newBuilder()
-                        .setVisibility(Protos.DiscoveryInfo.Visibility.FRAMEWORK)
+            if (!taskBuilder.hasDiscovery()) {
+                // Initialize with defaults:
+                taskBuilder.getDiscoveryBuilder()
+                        .setVisibility(Constants.DEFAULT_TASK_DISCOVERY_VISIBILITY)
```

Comment:
The YAML has an advertise bool but it's under the VIP. It's removed in this PR as it's not actually being used anywhere (it defaults to 'true' anyway, which is what you'd want if you were going to the trouble of creating a VIP). It's going to be promoted to the Port in a following behavior-changing PR.